app/Image/BaseImageHandler.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Image;
+
+use App\Exceptions\MediaFileOperationException;
+use App\Models\Configs;
+use App\Models\Logs;
+use Spatie\LaravelImageOptimizer\Facades\ImageOptimizer;
+
+abstract class BaseImageHandler implements ImageHandlerInterface
+{
+	/** @var int sentinel value to indicate that the used-defined compression quality should be loaded from runtime configuration */
+	public const USER_DEFINED_COMPRESSION_QUALITY = 0;
+
+	/** @var int default compression quality which is used when no other compression quality is set */
+	public const DEFAULT_COMPRESSION_QUALITY = 90;
+
+	/** @var int the desired compression quality, only used for JPEG during save */
+	protected int $compressionQuality = self::DEFAULT_COMPRESSION_QUALITY;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function __construct(int $compressionQuality = self::USER_DEFINED_COMPRESSION_QUALITY)
+	{
+		$this->compressionQuality = $compressionQuality === self::USER_DEFINED_COMPRESSION_QUALITY ?
+			Configs::getValueAsInt('compression_quality') :
+			$compressionQuality;
```

Comment:
I don't understand the purpose of this all. Why can't we simply rely on the value from `compression_quality`?

Reply:
Fixed by https://github.com/LycheeOrg/Lychee/pull/1351/commits/641a0669f70dcd883743cef9e26398997c4b9fa8